chore: merge dev into main

JoshuaWell opened the pull request:
# Summary

- merge dev into main

# Checklist

- [x] You agree with our [CLA](https://gist.githubusercontent.com/emdgroup-admin/16cc45ea4315c2ef29eb9d9afc36fcf5/raw/abb9c91f15278a62b9ac3e66144bcd27fa9485c2/CLA.md)
- [x] Included tests (or is not applicable).
- [x] Updated documentation (or is not applicable).

JoshuaWellbrock committed:
fix!: remove adding commands to queue as they need to be wrapped in a device specific command wrapper
BREAKING CHANGE: all methods in CmdWrapper now return the UrpCoreCommand

JoshuaWellbrock committed:
chore: add tests

JoshuaWellbrock committed:
fix: remove ChargingStationCmdWrapper

JoshuaWell committed:
Merge pull request #1 from emdgroup/fix_core_commands

fix!: core commands

KochC committed:
trigger-ci manually

JoshuaWellbrock committed:
Revert "fix!: remove adding commands to queue as they need to be wrapped in a device specific command wrapper"

This reverts commit bdc12b041f825b1dcb304117c46f44bff329dd5f.

JoshuaWellbrock committed:
fix: remove adding commands to queue as they need to be wrapped in a device specific command wrapper

JoshuaWellbrock committed:
chore: fix linting issues

JoshuaWell committed:
Merge pull request #2 from emdgroup/fix_commands

fix: remove adding commands to queue as they need to be wrapped in a …

Commit:
chore(release): 8.0.2-0

Commit:
chore(release): 8.0.2-0 [skip ci]

Commit:
chore(release): 8.0.2-0

KochC committed:
Update publish_dev.yaml [skip ci]

KochC committed:
Update package.json

Commit:
chore(release): 8.0.2-2

Commit:
chore(release): 8.0.2-2 [skip ci]

Commit:
chore(release): 8.0.2-2

JoshuaWellbrock committed:
fix!: BREAKING CHANGE: change CmdWrapper to abstract class

KochC committed:
Merge pull request #3 from emdgroup/fix_cmd_wrapper

fix!: Abstract CmdWrapper

Commit:
chore(release): 9.0.0-0

Commit:
chore(release): 9.0.0-0

Commit:
chore(release): 9.0.0-0 [skip ci]

KochC committed:
Update publish_dev.yaml [skip ci]

KochC committed:
feat: add package_checker script and integrate into publish workflow

KochC committed:
Merge pull request #4 from emdgroup/fix--pub.dev-delay-workaround

fix: add package_checker script and integrate into publish workflow [skip ci]

ElectricCookie committed:
chore: bump to liquid 18 & improve developer experience (#5)

* chore: bump to liquid 18
* refactor: use withAlpha for opacities as withOpacity is deprecated
* chore: update goldens
* chore: ignore build folder
* chore: consolidate scripts
* chore: use args package for packages.dart
* chore: add husky pre-commit hook
* chore: remove old pipeline file
* chore: adapt workflows to use adjusted packages.dart file
* chore: fix usage of --source option
* chore: update gh actions email to emdgroup domain
* chore: remove old util script
* chore: update dependencies
* chore: update Readme
* chore: fix usage of packages.dart
* chore: fix pr_dev to use packages.dart correctly
* chore: remove launch.json file
* chore: adjust golden threshold to 1%
* ci: fix the dart run packages.dart set-deps --source=hosted call in the build pipeline
* chore: bump to liquid flutter 18.2.0 to fix modal aspect ratio

Commit:
chore(release): 9.0.0-1

Commit:
chore(release): 9.0.0-1

Commit:
chore(release): 9.0.0-1 [skip ci]

KochC committed:
Update publish_dev.yaml

Commit:
chore(release): 9.0.0-2

Commit:
chore(release): 9.0.0-2

Commit:
chore(release): 9.0.0-2 [skip ci]

KochC committed:
Update publish_dev.yaml

Commit:
chore(release): 9.0.0-3

Commit:
chore(release): 9.0.0-3 [skip ci]

Commit:
chore(release): 9.0.0-3

KochC committed:
Update publish_dev.yaml

Commit:
chore(release): 9.0.0-4

Commit:
chore(release): 9.0.0-4

Commit:
chore(release): 9.0.0-4 [skip ci]

KochC committed:
Update publish_dev.yaml

Commit:
chore(release): 9.0.0-5

Commit:
chore(release): 9.0.0-5 [skip ci]

Commit:
chore(release): 9.0.0-5

ElectricCookie committed:
chore: Update dependencies and device connector modal styling

KochC committed:
Merge pull request #6 from emdgroup/feat_liquid_flutter_19

feat: liquid flutter 19

Commit:
chore(release): 9.0.0-6

Commit:
chore(release): 9.0.0-6 [skip ci]

Commit:
chore(release): 9.0.0-6

JoshuaWell committed:
feat: token (#7)

* feat: token
* fix: chunk ble messages
* test: add API test
* fix: device connector modal, add API_URL env variable
* fix: device connector modal, add API_URL env variable

---------

Co-authored-by: Joshua Wellbrock <info@joshuawellbrock.com>
Co-authored-by: KochC <Christopher.Koch@merckgroup.com>

KochC committed:
fix: validation of urp-core (#8)

* feat: token
* fix: chunk ble messages
* chore: bump versions
* chore: change api url
* test: add API test
* fix: device connector modal, add API_URL env variable
* fix: device connector modal, add API_URL env variable

---------

Co-authored-by: Joshua Wellbrock <info@joshuawellbrock.com>
Co-authored-by: Joshua Wellbrock <157679005+JoshuaWell@users.noreply.github.com>

Commit:
chore(release): 9.0.0-7

Commit:
chore(release): 9.0.0-7 [skip ci]

Commit:
chore(release): 9.0.0-7

JoshuaWell committed:
chore: bump urp types version (#9)

Co-authored-by: Joshua Wellbrock <info@joshuawellbrock.com>

Commit:
chore(release): 9.0.0-8

Commit:
chore(release): 9.0.0-8 [skip ci]

Commit:
chore(release): 9.0.0-8

JoshuaWellbrock committed:
feat: api exception

JoshuaWell committed:
Merge pull request #10 from emdgroup/feat_api_exception

feat: api exception

Commit:
chore(release): 9.0.0-9

Commit:
chore(release): 9.0.0-9 [skip ci]

Commit:
chore(release): 9.0.0-9

JoshuaWellbrock committed:
fix: API Exception

JoshuaWell committed:
Merge pull request #11 from emdgroup/fix_api_exception

fix: API Exception

Commit:
chore(release): 9.0.0-10

Commit:
chore(release): 9.0.0-10

Commit:
chore(release): 9.0.0-10 [skip ci]

JoshuaWellbrock committed:
fix: API Service status code

JoshuaWell committed:
Merge pull request #12 from emdgroup/fix_api

fix: API Service status code

Commit:
chore(release): 9.0.0-11

Commit:
chore(release): 9.0.0-11

Commit:
chore(release): 9.0.0-11 [skip ci]

JoshuaWellbrock committed:
fix: status code

JoshuaWell committed:
Merge pull request #13 from emdgroup/fix_status_code

fix: status code

Commit:
chore(release): 9.0.0-12

Commit:
chore(release): 9.0.0-12

Commit:
chore(release): 9.0.0-12 [skip ci]

JoshuaWellbrock committed:
chore: update deps

JoshuaWell committed:
Merge pull request #14 from emdgroup/chore_bump_versions

chore: bump urp_types version

Commit:
chore(release): 9.0.0-13

Commit:
chore(release): 9.0.0-13

Commit:
chore(release): 9.0.0-13 [skip ci]

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/emdgroup/mtrust-urp?pullRequest=15) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/emdgroup/mtrust-urp?pullRequest=15) before we can accept your contribution.<br/>**4** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: JoshuaWell<br/>:white_check_mark: JoshuaWellbrock<br/>:white_check_mark: KochC<br/>:white_check_mark: ElectricCookie<br/>:x: GitHub Actions<br/><hr/>**GitHub Actions** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/emdgroup/mtrust-urp?pullRequest=15) it.</sub>

JoshuaWellbrock committed:
chore: update readme

JoshuaWell committed:
Merge pull request #16 from emdgroup/fix_merge

chore: update readme

KochC committed:
feat: add commit-check job to GitHub Actions workflow

This update introduces a new job in the build_dev.yaml workflow that checks if the last commit was made by a user or GitHub Actions. The versioning job now depends on the success of this check, ensuring that it only runs if the last commit was made by a user.

KochC committed:
Merge pull request #17 from emdgroup/chore/ci_improvement
